internal/commands/buildpack_new.go

```diff
@@ -0,0 +1,93 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack"
+	"github.com/buildpacks/pack/internal/build"
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/logging"
+)
+
+// BuildpackNewFlags define flags provided to the BuildpackNew command
+type BuildpackNewFlags struct {
+	API     string
+	Path    string
+	Stacks  []string
+	Version string
+}
+
+// BuildpackCreator creates buildpacks
+type BuildpackCreator interface {
+	NewBuildpack(ctx context.Context, options pack.NewBuildpackOptions) error
+}
+
+// BuildpackNew generates the scaffolding of a buildpack
+func BuildpackNew(logger logging.Logger, client BuildpackCreator) *cobra.Command {
+	var flags BuildpackNewFlags
+	cmd := &cobra.Command{
+		Use:     "new <id>",
+		Short:   "Creates basic scaffolding of a buildpack.",
+		Args:    cobra.ExactValidArgs(1),
+		Example: "pack buildpack new sample/my-buildpack",
+		Long:    "buildpack new generates the basic scaffolding of a buildpack repository. It creates a new directory `name` in the current directory (or at `path`, if passed as a flag), and initializes a buildpack.toml, and two executable bash scripts, `bin/detect` and `bin/build`. ",
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			id := args[0]
+			idParts := strings.Split(id, "/")
+			dirName := idParts[len(idParts)-1]
+
+			var path string
+			if len(flags.Path) == 0 {
+				cwd, err := os.Getwd()
+				if err != nil {
+					return err
+				}
+				path = filepath.Join(cwd, dirName)
+			} else {
+				path = flags.Path
+			}
+
+			_, err := os.Stat(path)
+			if !os.IsNotExist(err) {
+				return errors.New(fmt.Sprintf("Directory exists! (%s)", path))
```

Comment:
nit: We typically format errors slightly different

```suggestion
				return errors.New(fmt.Sprintf("Directory %s exists!", style.Symbol(path)))
```

Reply:
I'll merge this. I need to update the test too though